### ISSUE TITLE
[BUG] Too early to record another clip

### ISSUE THREAD

**Issue by toshi2d** [marissa999/decky-recorder#52]
Found that when i'm using bluetooth headphones recorder is not working(notifications appear: too early to record another clip. When disconnecting headphones, recorder work's awesome.

**Comment by FoodFrisbee** [comment]
I'm getting this all the time, bluetooth or not

**Comment by raveensrk** [comment]
Is this issue fixed yet. I have steam deck 64gb. I have the same issues. I have soft linked /home/deck/Videos to microsd card.

**Comment by raveensrk** [comment]
Actually, it is working when I turn off the Bluetooth. Very strange. When bluetooth is on, it is not able to record.

**Comment by EliasGagnef** [comment]
not able to save a recording at all, even if Bluetooth is off

**Comment by spudpiggy** [comment]
^

**Comment by Conan179** [comment]
https://github.com/marissa999/decky-recorder/issues/62

**Comment by raveensrk** [comment]
@safijari is this fixed?

**Comment by spacecowboyx** [comment]
I have confirmed that disabling Bluetooth does allow the plugin to record manually and using snapshot mode. I don't know why Bluetooth being enabled makes it not work but it's disappointing if you are using bt headphones or a BT speaker when playing.

**Comment by safijari** [comment]
The Bluetooth capture issue has been fixed. I can deploy it but it creates the potential problem that switching between audio devices could leave you without audio. 

I'm still trying to figure that one out.

**Comment by spudpiggy** [comment]
Deploy it to beta.I don't need to switch audio devices most of the time, I just use internal speakers.Sent from my Galaxy
-------- Original message --------From: Jari ***@***.***> Date: 05/09/2023  14:49  (GMT+00:00) To: marissa999/decky-recorder ***@***.***> Cc: spudpiggy ***@***.***>, Comment ***@***.***> Subject: Re: [marissa999/decky-recorder] [BUG] Too early to record another clip (Issue #52) 
The Bluetooth capture issue has been fixed. I can deploy it but it creates the potential problem that switching between audio devices could leave you without audio.
I'm still trying to figure that one out.

**Comment by LuisMayo** [comment]
In the meantime, in my case at least (or maybe it's another bug), it's fixed if I stop and start replay mode

**Comment by spudpiggy** [comment]
Yeah, that's what I've been doing.Shouldn't be required thoSent from my Galaxy
-------- Original message --------From: Luis Mayo Valbuena ***@***.***> Date: 15/09/2023  09:50  (GMT+00:00) To: marissa999/decky-recorder ***@***.***> Cc: spudpiggy ***@***.***>, Comment ***@***.***> Subject: Re: [marissa999/decky-recorder] [BUG] Too early to record another clip (Issue #52) 
In the meantime, in my case at least (or maybe it's another bug), it's fixed if I stop and start replay mode

**Comment by lonkelle** [comment]
I get this a lot after waking from sleep. I just have to disable Recorder and Re-Enable it and it starts working again. 

It's annoying but makes me feel like this should hopefully be an easy solve.

**Comment by Toa-Nuva** [comment]
I also got this "too early to record..." error when in Replay mode. And during manual recording it did not show any errors, but it also did not save any video.
If this happens to anyone else, just make sure that you have the most recent versions installed of both SteamOS and of Decky Recorder. (In my case SteamOS was outdated, so updating that solved it for me.)

Seems that SteamOS includes GStreamer and Decky Recorder includes some additional libraries for that, and if there is a version mismatch between those, then recording will fail silently. Decky Recorder just starts the gst-launch-1.0 process (which fails immediately), and then it never checks the status of that process (until after you stop the recording), so it does not notice or report any issue.

**Comment by lonkelle** [comment]
@siglmf To confirm, "Disabling" and "Enabling" Decky Recorder didn't fix this for you (just checking if my error is different than yours)?

If this is the root of the error, then fetching those libraries dynamically would be a solve would it not. Assuming SteamOS allows for dynamic loaded libraries.

**Comment by siglmf** [comment]
I didn't see a way to disable and enable the whole plugin. *I think* I tried reloading the plugin (but I'm not 100% sure anymore :sweat_smile:) and I did toggle the Replay Mode on and off many times, but that did not fix the issue. And the issue also definitely was not related to Sleep mode, recording just did not work for me, at all.

Anyways, I guess I can provide a bit more information on this, as I did spend some time debugging this issue (by setting additional environment variables and debug printing in the main.py, so I did not have to rebuild the entire plugin myself :sweat_smile:). In the end I managed to get these warning messages in the `decky-recorder-std-err.log` file:
```
0:00:00.002044135  5821 0x55c7c9ffb800 WARN      GST_PLUGIN_LOADING gstplugin.c:491:gst_plugin_register_func: plugin "/home/deck/homebrew/plugins/decky-recorder/bin/gstreamer-1.0/libgstsbc.so" has incompatible version (plugin: 1.22, gst: 1.20), not loading
0:00:00.003126953  5821 0x55c7c9ffb800 WARN      GST_PLUGIN_LOADING gstplugin.c:491:gst_plugin_register_func: plugin "/home/deck/homebrew/plugins/decky-recorder/bin/gstreamer-1.0/libgstcodecalpha.so" has incompatible version (plugin: 1.22, gst: 1.20), not loading
0:00:00.004118218  5821 0x55c7c9ffb800 WARN      GST_PLUGIN_LOADING gstplugin.c:491:gst_plugin_register_func: plugin "/home/deck/homebrew/plugins/decky-recorder/bin/gstreamer-1.0/libgstaudiovisualizers.so" has incompatible version (plugin: 1.22, gst: 1.20), not loading
...
```
and so on, it was around 150 lines of similar messages. That's what pointed me to the version mismatch. Anyways, since GStreamer could not load these libraries, it did not recognize the `pipewiresrc` that is used to do the recording, so it basically just writes `WARNING: erroneous pipeline: no element "pipewiresrc"` to the logs and then quits.
